paper/src/main/java/com/denizenscript/denizen/paper/properties/PaperEntityExtensions.java

```diff
@@ -0,0 +1,172 @@
+        // For example, if the xp orb was spawned from breeding this would return the baby.
+        // -->
+        EntityTag.registerSpawnedOnlyTag(EntityFormObject.class, "xp_source", (attribute, entity) -> {
+            if (!(entity.getBukkitEntity() instanceof ExperienceOrb experienceOrb)) {
+                attribute.echoError("Entity " + entity + " is not an experience orb.");
+                return null;
+            }
+            UUID uuid = experienceOrb.getSourceEntityId();
+            if (uuid == null) {
+                return null;
+            }
+            Entity e = EntityTag.getEntityForID(uuid);
+            if (e == null) {
+                return null;
+            }
+            return new EntityTag(e).getDenizenObject();
+        });
+
+        // <--[tag]
+        // @attribute <EntityTag.spawn_location>
+        // @returns LocationTag
+        // @group properties
+        // @Plugin Paper
+        // @description
+        // Returns the initial spawn location of this entity.
+        // -->
+        EntityTag.tagProcessor.registerTag(LocationTag.class, "spawn_location", (attribute, entity) -> {
+            Location loc = entity.getBukkitEntity().getOrigin();
+            return loc != null ? new LocationTag(loc) : null;
+        });
+
+        // <--[tag]
+        // @attribute <EntityTag.from_spawner>
+        // @returns ElementTag(Boolean)
+        // @group properties
+        // @Plugin Paper
+        // @description
+        // Returns whether the entity was spawned from a spawner.
+        // -->
+        EntityTag.tagProcessor.registerTag(ElementTag.class, "from_spawner", (attribute, entity) -> {
+            return new ElementTag(entity.getBukkitEntity().fromMobSpawner());
+        });
+
+        // <--[mechanism]
+        // @object EntityTag
+        // @name goat_ram
+        // @input EntityTag
+        // @Plugin Paper
+        // @description
+        // Causes a goat to ram the specified entity.
+        // -->
+        EntityTag.registerSpawnedOnlyMechanism("goat_ram", false, EntityTag.class, (object, mechanism, input) -> {
+            if (object.getBukkitEntity() instanceof Goat) {
+                ((Goat) object.getLivingEntity()).ram(input.getLivingEntity());
+            }
+        });
+
+        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_19)) {
+
+            // <--[mechanism]
+            // @object EntityTag
+            // @name damage_item
+            // @input MapTag
+            // @Plugin Paper
+            // @description
+            // Damages the given equipment slot for the given amount.
+            // This runs all vanilla logic associated with damaging an item like gamemode and enchantment checks, events, stat changes, advancement triggers, and notifying clients to play break animations.
+            // Valid equipment slot values can be found at <@link url https://hub.spigotmc.org/javadocs/spigot/org/bukkit/inventory/EquipmentSlot.html>.

```

Comment:
Meta (for both mechs) needs to document the input map more specifically - what keys does it have an what are their values.
Can see meta for existing mechanisms that take `MapTag`s/ask on Discord if you'd like some examples